### PR TITLE
Serializable property

### PR DESCRIPTION
**Pull request by syuhei176**
Make property serializable. This enables the claim which smart contract can deserialize and makes channel on plasma and swap predicate more simple.

I'm waiting PR https://github.com/cryptoeconomicslab/ethabi/pull/10.

### Task List

- Removing PropertyFactory
- Property has address and PropertyInputs
- InputType is variable or constant value

### CODE REVIEW

File: ovm/src/types/input_type.rs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+use ethabi::{ParamType, Token};
+use ethereum_types::{Address, H256};
+use plasma_core::data_structure::abi::{Decodable, Encodable};
+use plasma_core::data_structure::error::{
+    Error as PlasmaCoreError, ErrorKind as PlasmaCoreErrorKind,
+};
+use plasma_core::data_structure::Range;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum InputType {
```
Comment (syuhei176) [inline ovm/src/types/input_type.rs]: I'll add a comment.

**Comment by syuhei176** [comment]
We can write claims without PropertyFactory.
https://github.com/cryptoeconomicslab/plasma-rust-framework/pull/208/files#diff-0bb2a15b985d7a1c86bd9bcf67ed303aR11-R24

One important modification is for-all-such-that decider. It set variable and decide children properties.
https://github.com/cryptoeconomicslab/plasma-rust-framework/pull/208/files#diff-001ed6daff371fc8bad2068d756df0d4R47-R74

**Comment by tkmct** [comment]
What does PropertyInput do?

**Comment by syuhei176** [comment]
> What does PropertyInput do?

I added PropertyInput to realize replacing variables.
For example, `PropertyInput::Placeholder` is for `$$message$$` in Will's comment.
https://github.com/plasma-group/ovm/issues/1#issuecomment-524421191

**Comment by syuhei176** [comment]
This is ownership property.
https://github.com/cryptoeconomicslab/plasma-rust-framework/pull/208/files#diff-d36d238a957767db2be9c1c8d09ad53fR10-R20

Any property using state_update placeholder can be adapted for Plasma.

**Pull request merged**